zip-0312.rst

```diff
@@ -1,7 +1,466 @@
+    Each element in the list indicates a NonZeroScalar identifier j and two commitment
+    Element values (hiding_nonce_commitment_j, binding_nonce_commitment_j).
+    This list MUST be sorted in ascending order by identifier.
+  - randomizer_point, an element in G (sent by the Coordinator).
+
+  Outputs:
+  - sig_share, a signature share, a Scalar.
+
+  def sign(identifier, sk_i, group_public_key, nonce_i, msg, commitment_list):
+    # Compute the randomized group public key
+    randomized_group_public_key = group_public_key + randomizer_point
+
+    # Compute the binding factor(s)
+    binding_factor_list = compute_binding_factors(commitment_list, msg, randomizer_point)
+    binding_factor = binding_factor_for_participant(binding_factor_list, identifier)
+
+    # Compute the group commitment
+    group_commitment = compute_group_commitment(commitment_list, binding_factor_list)
+
+    # Compute the interpolating value
+    participant_list = participants_from_commitment_list(commitment_list)
+    lambda_i = derive_interpolating_value(identifier, participant_list)
+
+    # Compute the per-message challenge
+    challenge = compute_challenge(group_commitment, randomized_group_public_key, msg)
+
+    # Compute the signature share
+    (hiding_nonce, binding_nonce) = nonce_i
+    sig_share = hiding_nonce + (binding_nonce * binding_factor) + (lambda_i * sk_i * challenge)
+
+    return sig_share
+
+
+Signature Share Verification and Aggregation
+''''''''''''''''''''''''''''''''''''''''''''
+
+The `aggregate` function is changed to incorporate the randomizer as follows: ::
+
+  Inputs:
+  - commitment_list =
+      [(j, hiding_nonce_commitment_j, binding_nonce_commitment_j), ...], a
+    list of commitments issued in Round 1 by each participant, where each element
+    in the list indicates a NonZeroScalar identifier j and two commitment
+    Element values (hiding_nonce_commitment_j, binding_nonce_commitment_j).
+    This list MUST be sorted in ascending order by identifier.
+  - msg, the message to be signed, a byte string.
+  - sig_shares, a set of signature shares z_i, Scalar values, for each participant,
+    of length NUM_PARTICIPANTS, where MIN_PARTICIPANTS <= NUM_PARTICIPANTS <= MAX_PARTICIPANTS.
+  - group_public_key, public key corresponding to the group signing key,
+  - randomizer, the randomizer Scalar.
+
+  Outputs:
+  - (R, z), a Schnorr signature consisting of an Element R and Scalar z.
+  - randomized_group_public_key, the randomized group public key
+
+  def aggregate(commitment_list, msg, sig_shares, group_public_key, randomizer):
+    # Compute the binding factors
+    binding_factor_list = compute_binding_factors(commitment_list, msg)
+
+    # Compute the group commitment
+    group_commitment = compute_group_commitment(commitment_list, binding_factor_list)
+
+    # Compute the challenge
+    randomized_group_public_key = group_public_key + G * randomizer
```

Comment:
```suggestion
    randomized_group_public_key = group_public_key + randomizer_point
```
(We've already computed `randomizer_point` as per my comment above.)

Reply:
impossible,